### DISCUSSION TITLE
[4.0] Include the locale in the "Language" site settings dropdown

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#11195] [Ideas] [answered]
I love that the "Language" setting for sites is using Selectize in Craft 4.0! 

However, it'd be great if options could display their actual locale in addition to the language name – like they do in Craft 3.

For example, there are several different flavours of Norwegian and having the locale displayed is helpful to easily tell which is `nb`, which is `nb-NO` etc.

Craft 3:

![CleanShot 2022-05-11 at 09 49 25](https://user-images.githubusercontent.com/298510/167797369-05c7338f-65cf-48f5-8723-5bec5f0553c3.png)

Craft 4:

![CleanShot 2022-05-11 at 09 52 15](https://user-images.githubusercontent.com/298510/167797568-86805c64-d5a6-4292-85ec-96e4d624bef7.png)

**Comment by brandonkelly** [comment, marked as answer]
Good call. I’ve replaced the languages’ native name hints with locale IDs, for Craft 4.1: (ffc95804767695757848c0399f876d692453e99e)

<img width="385" alt="A site’s Language setting, showing that each language option reveals its locale ID as a hint within the expanded menu." src="https://user-images.githubusercontent.com/47792/167858751-d74185f9-65cb-4303-b097-36506d1357c9.png">

It’s still possible to search for languages via their native names though:

<img width="389" alt="A site’s Language setting, showing that searching for “espanol” is filtering the selectable options to languages with “espanol” in their native names, despite showing “Spanish” in the UI." src="https://user-images.githubusercontent.com/47792/167859589-d624bceb-b90d-4556-a38e-639fac3f71ab.png">

Users’ Language and Formatting Locale preferences still show the native language names as before, as there’s a higher chance that they have the wrong language currently selected, and rely on the translation to find the right selection. However it’s now going to be possible to search by the locale IDs there:

<img width="438" alt="A user account’s Language preference, showing that searching for “nb” is limiting the results to “Norwegian Bokmål” despite that locale ID not appearing visibly within the option." src="https://user-images.githubusercontent.com/47792/167860367-9caf7cbe-30ee-4798-af33-1758253ae7ce.png">

**Reply by mmikkel** [reply]
Looks great, thanks!

**Reply by brandonkelly** [reply]
Craft 4.1 is out with this change ✨

**Reply by bencresty** [reply]
Love this change! And also the change from Craft 3 to 4! It was hard to find a language in v3, but that's solved now! :)
